### ISSUE TITLE
MAE is so big...

### ISSUE THREAD

**Issue by stoneshi-1999** [HumaticsLAB/GTM-Transformer#2]
Ur works are awesome , Thx for ur code❤, but when I train the GTM model , the **val_mae is much bigger than the results in ur paper** . How many epochs should I trian?

I only have 1 GPU : RTX 3060 Laptop ,  and I wana see more training details ,so I **changed the " check_val_every_n_epoch=1 " from 50**

I dont konw why **val_mae ( around 64 )**  is much bigger than the **results in ur paper (around 30 )**, but the loss is already going bigger . Hope u can help me , THX in advance!

![image](https://user-images.githubusercontent.com/70521393/155868398-c0751c50-5a35-4df1-905e-d78775a0e963.png)

![image](https://user-images.githubusercontent.com/70521393/155868442-33fb2616-066d-4429-b943-f9a1cc38413f.png)

**Comment by geriskenderi** [comment]
Hey, first of all, thanks for checking out our work. It is indeed really strange, that seems to be a normal WAPE value from the results we have gotten in our paper. We train our models for 200 epochs. If you keep trying to train the models with the same hyperparameters defined in the train.py arguments and still don't get good results, this might be a data issue. If the issue persists, I will try to replicate it and see what happens.

**Comment by stoneshi-1999** [comment]
THX for ur reply ❤! After 200 epoch , I got the results as follow, I feeling it's gonna be lower if I train more epoch. **Does val_mae represent the WAPE value?** I'm a little bit confused about the relationship among the **val_mae, WAPE , MAE** , THX in advance! 

![image](https://user-images.githubusercontent.com/70521393/155907085-e09830f5-fcbb-4928-aa11-1314c18c9bc6.png)


![image](https://user-images.githubusercontent.com/70521393/155907030-d9255a33-2ff0-4d3e-a1e3-da8b552bcdc8.png)

**Comment by geriskenderi** [comment]
Ok perfect, I will try to replicate your issue and let you know. For now this remains open. The val_mae represents the testing MAE value, we simply use the validation wrapper to be able to evaluate on the testing set after each (N) epoch. As for the relationship between the error values, there is no perfect rule, of course the lower one error is then the lower the other, the only difference is that one is relative (WAPE) and the other is absolute (MAE). The MAE values you are getting are definitely strange since the data should also be normalized. As I said before, I will try to investigate this issue during this week, please be patient. Thanks again for pointing this out!

**Comment by stoneshi-1999** [comment]
Thanks for ur detailed explaination 🙇‍❤ But this morning，I found I just trianed the FCN model instead of GTM-Transformer... And it's really werid that the MAE is even much more bigger when I train the GTM-Transformer model...

The MAE values of 199 epoch I commeted yesterday were from FCN_RUN1 ( 27-02-2022-17-16-54）. 

![544896308956986202](https://user-images.githubusercontent.com/70521393/155999160-d5a272b0-fbcc-46bb-970e-590168a096a9.jpg)

Now, it‘s GTM and still training , because I changed epoch from 200 to 1000 to see what's gonna happen.
epoch=0 ,val_mae=409.327 ; epoch=553 val_mae=58.146 , and the mae value is still decreasing.

![image](https://user-images.githubusercontent.com/70521393/156082220-31ee796f-3448-4331-9cad-30d91536e1b8.png)

**Comment by geriskenderi** [comment]
Hi there,

I tried to replicate your issue but after only 50 epochs of training GTM I get a MAE of 31.25, which is already better than all the other methods in Table 1. Here is a screenshot from my logs, where there seems to be a very regular decrease in loss and error:
![image](https://user-images.githubusercontent.com/25037198/156180581-b9b07135-992d-4a1e-b41e-a298627cb908.png)
Some minor changes with the latest commit, where I also added a requirements.txt file so that everyone can check the versions of the packages we have used to develop this code. 

Please make sure you are running with the default hyperparameters if you wish to replicate the results of the paper. Also, please make sure you haven't modified anything in the model or dataloading code by mistake. I tested the system on a single TITAN RTX so small changes due to different GPU architectures are possible, but I don't see any way to reach such high MAE values as in your case. I'll be closing this issue for now, thanks again for trying out the code, I hope you can manage to use it in your projects and obtain great results.
